.github/workflows/bench.yml

```diff
@@ -0,0 +1,62 @@
+name: Benchmark
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: ['opened', 'reopened', 'synchronize']
```

Comment:
Can we limit this to PRs with a label for right now as we try this out?

It would also be smart to limit this to PRs touching `**/crates/`, I don't feel like benchmarks are quite as critical as tests, so it's okay if we miss a small bit of coverage in order to save costs.